src/core/thread/mle.cpp

```diff
@@ -1733,6 +1739,10 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
             mNetif.GetActiveDataset().Set(activeTimestamp, aMessage, offset + sizeof(tlv), tlv.GetLength());
         }
     }
+    else
+    {
+        VerifyOrExit(update, ;);
```

Comment:
The optional in this case means that the sender may or may not include a Pending Timestamp TLV depending on whether or not it has a Pending Dataset.